lib/matplotlib/cm.py

```diff
@@ -177,23 +136,67 @@ def register(self, cmap, *, name=None, force=False):
             registered name. True supports overwriting registered colormaps
             other than the builtin colormaps.
         """
+        if not isinstance(cmap, colors.Colormap):
+            raise ValueError("You must pass a Colormap instance. "
+                             f"You passed {cmap} a {type(cmap)} object.")
+
         name = name or cmap.name
-        if name in self and not force:
-            raise ValueError(
-                f'A colormap named "{name}" is already registered.')
-        register_cmap(name, cmap.copy())
+        if name in self:
+            if not force and name in self._builtin_cmaps:
+                msg = f"Trying to re-register the builtin cmap {name!r}."
+                raise ValueError(msg)
+            elif not force:
+                raise ValueError(
+                    f'A colormap named "{name}" is already registered.')
+            else:
+                _api.warn_external(f"Trying to register the cmap {name!r} "
+                                   "which already exists.")
 
+        self._cmaps[name] = cmap.copy()
+
+    def unregister(self, name):
+        """
+        Remove a colormap from the registry.
+
+        You may not remove built-in colormaps.
+
+        If the named colormap is not registered, returns with no error, raises
+        if you try to de-register a default colormap.
+
+        .. warning::
+
+            The ColormapRegistry is a shared namespace that may be used
+            by multiple packages. Use `unregister` only if you know you
+            have registered that name before. In particular, do not
+            unregister just in case to clean the name before registering a
+            new colormap.
+
+        Parameters
+        ----------
+        name : str
+            The name of the colormap to be removed
+
+        Returns
+        -------
+        Colormap or None
+            If the colormap was registered, return it if not return `None`
+
+        Raises
+        ------
+        ValueError
+            If you try to remove a default built-in colormap.
+        """
+        if name in self._builtin_cmaps:
+            raise ValueError(f"cannot unregister {name!r} which is a builtin "
+                             "colormap.")
+        return self._cmaps.pop(name, None)
```

Comment:
I know this has been the interface before. But returning a colormap seems quite pointless. I cannot imagine a practical scenario in which I unregister and keep hold of the colormap to do something with it. Returning a bool seems more clear. Since we introduce this as a new API, we're free to make that change (but would need two extra lines in the original `unregister` to keep that backward-compatible).

Reply:
I agree, you can call `colormaps.get()` right before unregistering. I actually took it one step farther and said we don't even need a return at all. Happy to add it in if you think it is valuable, but seems even cleaner to avoid any return value.

Reply:
Further simplification is good. :+1: